### ISSUE TITLE
make it so that SpawnRequest is composed of ExecutorConfig and RouterConfig

### ISSUE THREAD

**Issue by pretentious7** [drifting-in-space/plane#161]
currently a SpawnRequest looks like so:
```rust
pub struct SpawnRequest {
  pub drone_id: DroneId,
  pub image: String,
  pub max_idle_secs: Duration,
  pub env: HashMap<String, String>,
  pub metadata: HashMap<String, String>
  pub credentials: Option<DockerCredentials>,
  pub resource_limits: ResourceLimits
}
```

Of these drone_id, metadata, and backend_id are needed for the scheduler, but the rest is essentially configuration for the docker executor. the api can remain fluid for now, but it seems prudent to have these seperated into an ExecutorConfig enum (or possibly a config method on the trait? not sure)

This will allow easy per spawn configuration, since all parameters can be saved in serialized form.

**Comment by paulgb** [comment]
Agreed. This is related to #137.

**Comment by paulgb** [comment]
Closed by #200
